Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressViewModel.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  SelectOwnerAddressViewModel.swift
+//  Multisig
+//
+//  Created by Moaaz on 10/8/20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import Combine
+import Web3
+
+class SelectOwnerAddressViewModel: ObservableObject {
+    @Published
+    var addresses = [Address]()
+    @Published
+    var selectedIndex = 0
+
+    private var rootNode: HDNode?
+    private var maxAddressesCount = 100
+    private var pageSize = 20
+
+    var canLoadMoreAddresses: Bool {
+        addresses.count < maxAddressesCount
+    }
+
+    init(rootNode: HDNode?, onSubmit: (() -> Void)? = nil ) {
+        self.rootNode = rootNode
+        generateAddressesPage()
+    }
+
+    private func addressAt(_ index: Int) -> Address? {
+        guard let pkData = rootNode?.derive(index: UInt32(index), derivePrivateKey: true)?.privateKey else {
+            return nil
+        }
+
+        do {
+            let address = try EthereumPrivateKey(hexPrivateKey: pkData.toHexString()).address
+            return Address(address, index: index)
```

Comment:
we decided to show indexes starting with 1; so you can use here `return Address(address, index: index + 1)`

Reply:
I have just made update, index will start from 0 but in UI will be displayed starting from 1